exec/cnex/cnex.c

```diff
@@ -28,28 +29,82 @@ typedef struct tagTCommandLineOptions {
     int  ArgStart;
     char *pszFilename;
     char *pszExecutableName;
+    char *pszExecutablePath;
 } TOptions;
 
-static TOptions gOptions = { FALSE, FALSE, TRUE, 0, NULL, NULL };
+static TOptions gOptions = { FALSE, FALSE, TRUE, 0, NULL, NULL, NULL };
 
-void exec_freeExecutor(TExecutor *e)
+static char *getApplicationName(char *arg)
 {
-    uint32_t i;
+    char *p = arg;
+    char *s = p;
+
+    while (*p) {
+        if (*p == '\\' || *p == '/') {
+            s = p;
+        }
+        p++;
+    }
+    return ++s;
```

Comment:
Using preincrement here is being unnecessarily tricky. It also doesn't work if the arg doesn't contain any slashes at all. Better to just `return s;` here, and use `s = p + 1;` above.

Reply:
Changed.

Reply:
This does not appear to have been changed. The original comment still applies.